### PR TITLE
[Bug] Fix thumbnail generation if asset or config is null

### PR DESCRIPTION
**Pull request by mattamon**
## Changes in this pull request  
Error images do not need an asset or config. Move checks to certain conditions

### CODE REVIEW

**Comment by blankse** [comment]
Target branch should be `11.2`

**Comment by mattamon** [comment]
> Target branch should be `11.2`

I know, but I want to test something first. Changes will be cherry picked anyways

File: models/Asset/Image/Thumbnail.php [hunk 1]
```diff
@@ -112,18 +112,20 @@ protected function useOriginalFile(string $filename): bool
...
-            throw new ThumbnailMaxScalingFactorException();
-        }
-
         $deferred = false;
         $generated = false;
 
         if ($this->asset && empty($this->pathReference)) {
+
+
+            if ($this->config && !$this->checkAllowedFormats($this->config->getFormat(), $this->asset)) {
```
Comment (blankse) [inline models/Asset/Image/Thumbnail.php:121]: Can be moved to the else block in line 136 without config check, right?

Comment (mattamon) [reply models/Asset/Image/Thumbnail.php:121]: I already tried that, dealing with "cognitive overload" and qodana currently, trying to resolve it :) but thanks

**Comment by mattamon** [comment]
Closed in favor of #17039

**Pull request closed without merging**